File: app/buildz/tmp/kapt3/stubs/debug/com/bits/bee/bpmc/data/data_source/local/dao/CstrDao.java
package com.bits.bee.bpmc.data.data_source.local.dao;

import java.lang.System;

/**
 * Created by aldi on 07/06/22.
 */
@androidx.room.Dao()
@kotlin.Metadata(mv = {1, 7, 1}, k = 1, d1 = {"\u0000\"\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0010\u000e\n\u0000\n\u0002\u0010\b\n\u0000\bg\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001J\n\u0010\u0003\u001a\u0004\u0018\u00010\u0002H\'J\u0018\u0010\u0004\u001a\u00020\u00052\u0006\u0010\u0006\u001a\u00020\u00072\u0006\u0010\b\u001a\u00020\tH\'\u00a8\u0006\n"}, d2 = {"Lcom/bits/bee/bpmc/data/data_source/local/dao/CstrDao;", "Lcom/bits/bee/bpmc/data/data_source/local/base/BaseDao;", "Lcom/bits/bee/bpmc/data/data_source/local/model/CstrEntity;", "getLastId", "updateCode", "", "code", "", "id", "", "app_debug"})
public abstract interface CstrDao extends com.bits.bee.bpmc.data.data_source.local.base.BaseDao<com.bits.bee.bpmc.data.data_source.local.model.CstrEntity> {
    
    @androidx.room.Query(value = "UPDATE cstr\nSET kode_cstr = :code\nWHERE id = :id")
    public abstract void updateCode(@org.jetbrains.annotations.NotNull()
    java.lang.String code, int id);
    
    @org.jetbrains.annotations.Nullable()
    @androidx.room.Query(value = "SELECT * FROM cstr ORDER BY id DESC LIMIT 1")
    public abstract com.bits.bee.bpmc.data.data_source.local.model.CstrEntity getLastId();
}

File: app/src/main/java/com/bits/bee/bpmc/data/data_source/local/dao/CstrDao.kt
package com.bits.bee.bpmc.data.data_source.local.dao

import androidx.room.Dao
import androidx.room.Query
import com.bits.bee.bpmc.data.data_source.local.base.BaseDao
import com.bits.bee.bpmc.data.data_source.local.model.CashEntity
import com.bits.bee.bpmc.data.data_source.local.model.CstrEntity
import com.bits.bee.bpmc.data.data_source.local.model.PossesEntity

/**
 * Created by aldi on 07/06/22.
 */
@Dao
interface CstrDao : BaseDao<CstrEntity>{
    @Query("UPDATE cstr\n" +
            "SET kode_cstr = :code\n" +
            "WHERE id = :id")
    fun updateCode(code: String, id: Int)

    @Query("SELECT * FROM cstr ORDER BY id DESC LIMIT 1")
    fun getLastId(): CstrEntity?
}